pkg/operator/sync_configmapcabundle_v311_00.go

```diff
@@ -0,0 +1,130 @@
+package operator
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsclientv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	operatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	scsv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcecread"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/service-serving-cert-signer/pkg/operator/v310_00_assets"
+)
+
+// syncConfigMapCABundleController_v311_00_to_latest takes care of synchronizing (not upgrading) the thing we're managing.
+// most of the time the sync method will be good for a large span of minor versions
+func syncConfigMapCABundleController_v311_00_to_latest(c ServiceCertSignerOperator, operatorConfig *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity) (operatorsv1alpha1.VersionAvailablity, []error) {
```

Comment:
All of this stuff in `pkg/operator/...` does not make sense to me. Either this update/install logic should be handled by the OLM or cluster operator. OR if there is special logic needed beyond applying a set of resources, it should be made into an operator proper, with a CRD and reworked to fit with the new installer and OLM model.

Reply:
This is essentially the same sync logic as the other two controllers managed by the operator. If we need to break the operator sync out to OLM/cluster operator then we should do that as a follow up. My main focus on this PR is the configmap injector functionality.